src/Frontend/Page/User.hs

```diff
@@ -96,7 +96,7 @@ userSettings = redirectFormHandler (PageUserSettings <$> currentUser) changeUser
   where
     -- FIXME: Set the password
     changeUser (UserSettingData email _oldPass _newPass1 _newPass2) = do
-        modifyCurrentUser (maybe id (\ e -> userEmail .~ Just e) email)
+        maybe nil (modifyCurrentUser (ModifyUserSetEmail e)) email
```

Comment:
I would prefer to move this `changeUser` function to Persistent.Pure, call it `setUser` and make it an update instead of changing modifyUser.